frontend/packages/console-shared/src/hooks/useTelemetry.ts

```diff
@@ -29,10 +37,53 @@ export const updateServerFlagsFromTests = () => {
 export const useTelemetry = () => {
   // TODO use useDynamicPluginInfo() hook to tell whether all dynamic plugins have been processed
   // to avoid firing telemetry events multiple times whenever a dynamic plugin loads asynchronously
+
+  const [currentUserPreferenceTelemetryValue] = useUserSettings<USER_TELEMETRY_ANALYTICS>(
+    PREFERRED_TELEMETRY_USER_SETTING_KEY,
+    null,
+    true,
+  );
+
   const [extensions] = useResolvedExtensions<TelemetryListener>(isTelemetryListener);
 
   return React.useCallback<TelemetryEventListener>(
     (eventType, properties) => {
+      if (
+        !window.SERVER_FLAGS.telemetry?.STATE ||
```

Comment:
remove this condition, if it's not set, we expect that telemetry on (starting with 4.16 at least)